earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -26,11 +29,42 @@
 using namespace std;
 using namespace getime;
 
-// Initialize static members of Profiler class
-PerformanceLogger * const PerformanceLogger::_instance = new PerformanceLogger();
+namespace performance_logger {
```

Comment:
I'm not super happy with the duplication of the locking code but I don't see a better way to handle this. Maybe include a comment about why it was duplicated.

Reply:
nvm, I see you added the comment in the .cpp file.